src/hotspot/cpu/x86/c1_Runtime1_x86.cpp

```diff
@@ -1111,7 +1111,19 @@ OopMapSet* Runtime1::generate_code_for(StubID id, StubAssembler* sasm) {
       }
 
       break;
-
+    case load_klass_id:
+#ifdef _LP64
+      {
+        StubFrame f(sasm, "load_klass", dont_gc_arguments);
+        sasm->save_live_registers_no_oop_map(true);
+        f.load_argument(0, c_rarg0); // obj
+        __ call_VM_leaf(CAST_FROM_FN_PTR(address, oopDesc::load_klass_runtime), c_rarg0);
+        sasm->restore_live_registers_except_rax(true);
+      }
+#else
+      __ should_not_reach_here();
```

Comment:
So the intent here to generate x86_32 stub, but check it is not called ever, right? I wonder if we still need to do `StubFrame f(sasm, "load_klass", dont_gc_arguments);` even for a simple `__ should_not_reach_here();`...

Reply:
Perhaps not. Let me see if I can avoid this altogether.